overlord/ifacestate/apparmorprompting/prompting.go

```diff
@@ -49,14 +49,14 @@ var (
 // A Manager holds outstanding prompts and mediates their replies, further it
 // stores and applies persistent rules.
 type Manager interface {
-	Prompts(userID uint32) ([]*requestprompts.Prompt, error)
-	PromptWithID(userID uint32, promptID prompting.IDType) (*requestprompts.Prompt, error)
-	HandleReply(userID uint32, promptID prompting.IDType, constraints *prompting.Constraints, outcome prompting.OutcomeType, lifespan prompting.LifespanType, duration string) ([]prompting.IDType, error)
+	Prompts(userID uint32, clientActivity bool) ([]*requestprompts.Prompt, error)
+	PromptWithID(userID uint32, promptID prompting.IDType, clientActivity bool) (*requestprompts.Prompt, error)
+	HandleReply(userID uint32, promptID prompting.IDType, constraints *prompting.Constraints, outcome prompting.OutcomeType, lifespan prompting.LifespanType, duration string, clientActivity bool) ([]prompting.IDType, error)
```

Comment:
if you plan to add more arguments, then perhaps it's useful to pass a flags struct here (also avoids the problem of having 'bare' bools):
```
type Flags struct {
    UserAction bool
}

// but pass it as value
type Manager interface {
	...
	HandleReply(userID uint32, promptID prompting.IDType, constraints *prompting.Constraints, 
		outcome prompting.OutcomeType, lifespan prompting.LifespanType, duration string, 
		flags Flags,
	) ([]prompting.IDType, error)
}
```